matter_server/common/models/node.py

```diff
@@ -0,0 +1,133 @@
+    # device_type_instances: list[MatterDeviceTypeInstance] = field(default_factory=list)
+    # node_devices: list[AbstractMatterNodeDevice] = field(default_factory=list)
+
+    def parse_attributes(self, attributes: dict) -> None:
+        """Try to parse a MatterNode from AttributeCache (retrieved from Read)."""
+
+        device_type_instances: list[MatterDeviceTypeInstance] = []
+
+        for endpoint_id, clusters in attributes.items():
+            for cluster_cls, cluster in clusters.items():
+
+                if isinstance(cluster, Attribute.ValueDecodeFailure):
+                    # yes, this may happen
+                    continue
+
+                # the python wrapped SDK has a funky way to index the Clusters by
+                # having the class itself as dict key. We change that here to just the Cluster ID.
+                self.attributes[cluster.id] = cluster
+
+                # lookup device type for this cluster
+                device_type = ALL_TYPES.get(cluster.id)
+
+                if device_type is None:
+                    LOGGER.warning("Found unknown device type for Cluster %s", cluster)
+                    continue
+
+                instance = MatterDeviceTypeInstance(
+                    self, device_type, int(endpoint_id), cluster.clusterRevision
+                )
+                
+
+    def parse_attributes_org(self, attributes: dict) -> None:
+        """Try to parse a MatterNode from AttributeCache (retrieved from Read)."""
+
+        device_type_instances: list[MatterDeviceTypeInstance] = []
+
+        for endpoint_id, clusters in attributes.items():
+            for cluster_cls, cluster in clusters.items():
+
+                if isinstance(cluster, Attribute.ValueDecodeFailure):
+                    # yes, this may happen
+                    continue
+
+                # the python wrapped SDK has a funky way to index the Clusters by
+                # having the class itself as dict key. We change that here to just the Cluster ID.
+                self.attributes[cluster.id] = cluster
+
+                # lookup device type for this cluster
+                device_type = ALL_TYPES.get(cluster.id)
+
+                if device_type is None:
+                    LOGGER.warning("Found unknown device type for Cluster %s", cluster)
+                    continue
+
+                instance = MatterDeviceTypeInstance(
+                    self, device_type, int(endpoint_id), cluster.clusterRevision
+                )
+                if device_type is RootNode:
+                    self.root_device_type_instance = instance
+                elif device_type is Aggregator:
+                    self.aggregator_device_type_instance = instance
+                else:
+                    device_type_instances.append(instance)
+
+            self.device_type_instances = device_type_instances
+
+            if not hasattr(self, "root_device_type_instance"):
```

Comment:
It's not good to initialize attributes dynamically after init. Why do we not set this attribute in init?

Reply:
This is also just copied from the previous implementation and I did not have a chance to fix it

Reply:
left out for now. will fix in follow up while working on core part as I'm not 100% how to do the discovery properly